autoscaling.tf

```diff
@@ -1,6 +1,6 @@
 locals {
   alb_enabled                         = length(var.alb_name) > 0
-  mpr_enabled                         = var.mpr_enabled == true ? true : (lookup(module.this.tags, "Type") == "subscriber" || lookup(module.this.tags, "Type") == "consumer") && var.mpr_enabled == null ? true : false
+  mpr_enabled                         = var.mpr_enabled == true ? true : (module.this.tags["Type"] == "subscriber" || module.this.tags["Type"] == "consumer") && var.mpr_enabled == null ? true : false
```

Comment:
I think we might encounter with some cases like `module.this.tags["Type"]` doesn't exist at all, so instead of using in that way wouldn't it better if we use `try` function in here?

Reply:
if `module.this.tags["Type"]` doesn't exist, that means we are missing something in our resource definition and we'd get error saying it is missing. with `try` we are kind of skipping that error instead

Reply:
And also we always used `lookup` for every time we want to get "Type" key so I believe that we should be following the same approach.

Reply:
lookup does the same thing as tag["Type"]. the reason I am switching from lookup to map index is that lookup is deprecated. more details here: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/main/docs/rules/terraform_deprecated_lookup.md 

I am open to suggestions about  where and how to check the existence of `Type` tag and ensuring it always exists.

Reply:
lookup is not deprecated, omitting default value in lookup deprecated since v0.7

ref: https://developer.hashicorp.com/terraform/language/functions/lookup

Reply:
the way the lookup is used here is deprecated (without third default argument). it is only used for accessing the index of the map with only two args (map and index)